athena-dynamodb/src/main/java/com/amazonaws/athena/connectors/dynamodb/DynamoDBRecordHandler.java

```diff
@@ -134,21 +136,55 @@ protected void readWithConstraint(BlockSpiller spiller, ReadRecordsRequest recor
         // use the property instead of the request table name because of case sensitivity
         String tableName = split.getProperty(TABLE_METADATA);
         invokerCache.get(tableName).setBlockSpiller(spiller);
-        Iterator<Map<String, AttributeValue>> itemIterator = getIterator(split, tableName, recordsRequest.getSchema());
         DDBRecordMetadata recordMetadata = new DDBRecordMetadata(recordsRequest.getSchema());
+
+        String disableProjectionAndCasingEnvValue = System.getenv().getOrDefault(DISABLE_PROJECTION_AND_CASING_ENV, "auto").toLowerCase();
+        logger.info(DISABLE_PROJECTION_AND_CASING_ENV + " environment variable set to: " + disableProjectionAndCasingEnvValue);
+
+        boolean disableProjectionAndCasing = false;
+        if (disableProjectionAndCasingEnvValue.equals("auto")) {
+            // In the automatic case, we will try to mimic the behavior prior to the support of `set` and `decimal` types as much
```

Comment:
sorry to nit here, but why not a multi-line comment for this section?

Reply:
There are no style guidelines dictating one vs the other here.

You could imagine that I could have started out with a multi line comment and then someone else asks me to use // comments instead (which has happened before in other projects at other companies).

Reply:
okay, in that case, no problem.